src/Moryx.AbstractionLayer.Products.Endpoints/ProductManagementController.cs

```diff
@@ -0,0 +1,244 @@
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("[action]")]
+        public ActionResult<ProductInstance> GetInstance (long id)
+        {
+            if (id == 0)
+                return BadRequest($"Id was 0");
+            var productInstance = _productManagement.GetInstance(id);
+            if (productInstance == null)
+                return NotFound();
+            return Ok(productInstance);
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("[action]")]
+        public ActionResult<ProductInstance> GetInstance (IIdentity identity)
+        {
+            if (identity == null)
+                return BadRequest($"Identity was null");
+            var productInstance = _productManagement.GetInstance(identity);
+            if (productInstance == null)
+                return NotFound();
+            return Ok(productInstance);
+        }
+
+        [HttpGet]
+        [Route("[action]")]
+        public IReadOnlyList<ProductInstance> GetInstances(long [] ids)
+        {
+            return _productManagement.GetInstances(ids);
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        [Route("[action]")]
+        public ActionResult<IRecipe> GetRecipe(long id)
+        {
+            if (id == 0)
+                return BadRequest($"Id was 0");
+            var recipe = _productManagement.LoadRecipe(id);
+            if (recipe == null)
+                return NotFound();
+            return Ok(recipe);
+        }
+
+        [HttpGet]
+        [Route("[action]")]
+        public IReadOnlyList<Type> GetProductTypes()
+        {
+            return _productManagement.ProductTypes;
+        }
+
+        [HttpGet]
+        [Route("[action]")]
+        public IReadOnlyList<Type> GetRecipeTypes()
+        {
+            return _productManagement.RecipeTypes;
+        }
+        #endregion
+
+        #region Post-Requests
+
+        [HttpPost]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        [Route("[action]")]
+        public ActionResult<IProductType> Duplicate(IProductType template, string newIdentity)
+        {
+            if (template == null)
+                return BadRequest($"Template was null");
+            if (newIdentity == null)
+                return BadRequest($"New Identity was null");
+            var identityArray = newIdentity.Split('-'); 
+            var identity = new ProductIdentity(identityArray[0], Convert.ToInt16(identityArray[1]));
```

Comment:
Please add a call to WebUtility.EncodeHtml around identityArray[0] and a short.TryParse around identityArray[1] with a badrequestresult on a failed parse 😊